### ISSUE TITLE
Selecting all items doesn't update handlebars

### ISSUE THREAD

**Issue by smayesdxc** [microsoft-search/pnp-modern-search#2768]
**Version used**
4.8.0

**Describe the bug**
In a Search Results web part's Details List, when selecting all items and when using a custom handlebars results template, the array at `@root.selectedKeys` is not updated properly. This causes unexpected behavior. The array is not updated with the keys for all the items on the page and is left alone to the value it had prior to the user selecting all items. When using `@root.selectedKeys`, typically with the `isItemSelected` helper, this causes items to still appear/function as unselected even though they have been selected per the Details List control.

Unselecting all items has the same effect. The `@root.selectedKeys` array is not updated and retains whatever value it previously had.

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Configure a SharePoint modern page with a Search Results web part
2. Ensure that the `Allow items selection` and `Allow multiple selection` settings are both set to `On`
3. Use the Details List layout for the web part, then edit the results template and add a bit of custom handlebars code using `@root.selectedKeys` and the `isItemSelected` template. For example -

```handlebars
<div class="template">
    <!-- Beginning of custom handlebars code -->
    <div>
        <div>Selected items</div>
        {{#each data.items as |item|}}
            {{#if (isItemSelected @root.selectedKeys @index)}}
                <div>{{slot item @root.slots.Title}}</div>
            {{/if}}
        {{/each}}
    </div>
    <!-- End of custom handlebars code -->

    {{#if @root.properties.showSelectedFilters}}
        <pnp-selectedfilters 
            data-filters="{{JSONstringify filters.selectedFilters 2}}" 
            data-filters-configuration="{{JSONstringify filters.filtersConfiguration 2}}" 
            data-instance-id="{{filters.instanceId}}"
            data-operator="{{filters.filterOperator}}"
            data-theme-variant="{{JSONstringify @root.theme}}"
        >
        </pnp-selectedfilters>
    {{/if}}

    <!-- etc... -->
</div>
```

4. Run any search that returns multiple items
5. Select any single item
6. Click the select all checkbox in the header row which causes all rows in the current page to be selected. Notice that the custom code still only outputs the single item that was selected in Step 4 above.
7. Click the select all checkbox a second time in the header row which causes all rows in the current page to be unselected. Notice that the custom code still continues to output the single item that was selected in Step 4 above.

**Expected behavior**
I would expect the `@root.selectedKeys` array to contain all keys for all selected items, even when the select all feature is used. This would cause the handlebars `isItemSelected` helper to function properly also.

**Screenshots**

*After Step 4, with no items selected.*
![image](https://user-images.githubusercontent.com/12766851/218553067-6fb61fea-421e-477d-9552-e026cf7cd116.png)

*After Step 5, with one item selected.*
![image](https://user-images.githubusercontent.com/12766851/218553222-a9cfaa12-7119-444e-822c-aff9cec2c483.png)

*After Step 6, with all items selected. However, only one continues to show under the text `Selected items`.*
![image](https://user-images.githubusercontent.com/12766851/218553548-6745a7f7-e5fe-4d97-bc54-801d09bfb330.png)

*After Step 7, with all items de-selected. However, one still continues to show under the text `Selected items`.*
![image](https://user-images.githubusercontent.com/12766851/218553837-7b6621b1-eef6-44d8-b3b5-9e6ba64de54a.png)

**Desktop (please complete the following information):**
- Browser: Microsoft Edge 110.0.1587.41, 64-bit, running on Windows 10 Version 21H2 (Build 19044.2486)

**Additional context**
This has been observed in multiple tenants/sites.

**Comment by patrikhellgren** [comment]
@smayesdxc This issue is fixed by #2678 that will be available in the next release.

**Comment by smayesdxc** [comment]
Thank you @patrikhellgren - that's great news!
